gnomad_constraint/pipeline/constraint_pipeline.py

```diff
@@ -0,0 +1,132 @@
+                        hl.parse_locus_interval(c, reference_genome=rg)
+                        for c in [rg.contigs[19], rg.x_contigs[0], rg.y_contigs[0]]
+                    ]
+                    logger.info(
+                        "Filtering the %s HT to chr20, chrX, and chrY for testing...",
+                        data_type,
+                    )
+                    ht = hl.filter_intervals(ht, contigs_keep)
+
+                # Add annotations from VEP context Table to genome and exome Tables.
+                ht = (
+                    add_vep_context_annotations(ht, context_ht)
+                    if data_type != "context"
+                    else ht
+                )
+
+                # Filter input Table and add annotations used in constraint calculations.
+                full_ht = prepare_ht_for_constraint_calculations(ht)
+                # Filter to locus that is on an autosome or in a pseudoautosomal region.
+                ht = full_ht.filter(full_ht.locus.in_autosome_or_par())
+                ht.write(
+                    get_preprocessed_ht(data_type, version, "autosome_par", test).path,
+                    overwrite=overwrite,
+                )
+
+                # Sex chromosomes are analyzed separately, since they are biologically different from the autosomes.
+                if data_type != "genomes":
+                    filter_x_nonpar(full_ht).write(
+                        get_preprocessed_ht(
+                            data_type, version, "chrx_nonpar", test
+                        ).path,
+                        overwrite=overwrite,
+                    )
+                    filter_y_nonpar(full_ht).write(
+                        get_preprocessed_ht(
+                            data_type, version, "chry_nonpar", test
+                        ).path,
+                        overwrite=overwrite,
+                    )
+            logger.info("Done with preprocessing genome and exome Table.")
+
+    finally:
+        logger.info("Copying log to logging bucket...")
+        hl.copy_log(get_logging_path("constraint_pipeline"))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--overwrite", help="Whether to overwrite output files", action="store_true"
+    )
+    parser.add_argument(
+        "--version",
+        help=f"Which version of the resource Tables will be used. Default is {CURRENT_VERSION}.",
+        type=str,
+        default=CURRENT_VERSION,
+    )
+    parser.add_argument(
+        "--test",
+        help="Whether to filter the exome Table, genome Table and the context Table to only chromosome 20, chromosome X, and chromosome Y for testing.",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--preprocess-data",
+        help="Whether to prepare the exome, genome, and context Table for constraint calculations by adding necessary coverage, methylation level, and VEP annotations.",
```

Comment:
Sorry, I'm starting to try and make my strings fit the same length as we use for code using black. ```black --preview gnomad_constraint```
```suggestion
        help=(
            "Whether to prepare the exome, genome, and context Table for constraint"
            " calculations by adding necessary coverage, methylation level, and VEP"
            " annotations."
        ),
```

Reply:
I will add this to the `.pre-commit-config.yaml` as well!